impl/src/main/java/io/serverlessworkflow/impl/DefaultTaskExecutorFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020-Present The Serverless Workflow Specification Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.serverlessworkflow.impl;
+
+import io.serverlessworkflow.api.types.CallTask;
+import io.serverlessworkflow.api.types.Task;
+import io.serverlessworkflow.api.types.TaskBase;
+
+public class DefaultTaskExecutorFactory implements TaskExecutorFactory {
+
+  protected DefaultTaskExecutorFactory() {}
+
+  private static TaskExecutorFactory instance = new DefaultTaskExecutorFactory();
+
+  public static TaskExecutorFactory get() {
```

Comment:
This has been made public, allowing other implementations to reuse the code executing task as desired.